Используя информацию по мидиям Петербурга и Магадана, которые представлены в виде двух массивов petersburg и magadan в разделе ANOVA тест, проверьте данные на нормальность и на наличие корреляции. Для этого объедините 2 массива в DataFrame. Обоснуйте выбор теста на корреляцию. Сделайте вывод по гипотезе.

Также ответьте на вопрос:
Какова разница в среднем размере мидии в зависимости от города-производителя. Обоснуйте выбор теста. Сделайте вывод по гипотезе.


In [70]:
import pandas as pd
#Объединение два массива в DataFrame
df = pd.DataFrame({'Petersburg':[0.0974, 0.1352, 0.0817, 0.1016, 0.0968, 0.1064, 0.105], 
                   'Magadan':[0.1033, 0.0915, 0.0781, 0.0685, 0.0677, 0.0697, 0.0764]})
df

Petersburg  Magadan
0      0.0974   0.1033
1      0.1352   0.0915
2      0.0817   0.0781
3      0.1016   0.0685
4      0.0968   0.0677
5      0.1064   0.0697
6      0.1050   0.0764

In [71]:
H0 = 'Данные распределены нормально'
Ha = 'Данные не распределены нормально (мы отвергаем H0)'
alpha = 0.05
from scipy.stats import shapiro
_, p = shapiro(df)
print('p=%.3f' % p)
# Интерпретация 
if p > alpha:
	print(H0)
else:
	print(Ha)

p=0.226
Данные распределены нормально


Выбор теста на корреляцию:
1. Постановка вопроса: как изменяется размер раковины мидии (числовой признак) в зависимости от места выращивания? 
2. Выбор типа (теста) корреляции: так как данные распределены нормально, то метод Пирсона! 

In [72]:
import statistics
df.corr()['Petersburg']['Magadan']

0.25509468576821276

Вывод: очень слабая связь корреляции (зависимость) размера раковин мидий от места их выращивания.

Выбор теста на сравнение выборок:
1. Постановка вопроса: как влияет город (категориальная независимая переменная) на средний размер (зависимая числовая переменная) раковин мидий ?
2. Парный т-тест!

In [73]:
from scipy.stats import ttest_ind
H0 = 'Нет значимой разницы между размеров раковин мидий, выращенных в разных городах.'
Ha = 'Есть значимая разница между размерами раковин мидий, выращенных в разных городах.'
def t_test(df):
    print('\n' + "*** Результаты независимого T-теста ***")
    test_results = ttest_ind(df['Petersburg'], df['Magadan'], equal_var=True)

    p = round(test_results[1],2)

    if p>alpha:
        print(f"{p} > {alpha}. Мы не можем отвергнуть нулевую гипотезу. {H0}")
    else:
        print(f"{p} <= {alpha}. Мы отвергаем нулевую гипотезу. {Ha}")

t_test(df)


*** Результаты независимого T-теста ***
0.01 <= 0.05. Мы отвергаем нулевую гипотезу. Есть значимая разница между размерами раковин мидий, выращенных в разных городах.


Вывод по гипотезе: Мы отвергаем нулевую гипотезу. Есть значимая разница между средним размером раковины мидий в двух разных местах

In [74]:
df.astype(float)
df['Petersburg'].mean() - df['Magadan'].mean()

0.02412857142857143

Разница в среднем размере мидии в зависимости от города-производителя: 0.02412857142857143 m - 2,4 см около 1 дюйма